src/dask_awkward/lib/optimize.py

```diff
@@ -117,26 +118,14 @@ def _prepare_buffer_projection(
 
     hlg = HighLevelGraph(projection_layers, dsk.dependencies)
 
-    # this loop builds up what are the possible final leaf nodes by
-    # inspecting the dependents dictionary. If something does not have
-    # a dependent, it must be the end of a graph. These are the things
-    # we need to compute for; we only use a single partition (the
-    # first). for a single collection `.compute()` this list will just
-    # be length 1; but if we are using `dask.compute` to pass in
-    # multiple collections to be computed simultaneously, this list
-    # will increase in length.
-    leaf_layers_keys = [
-        (k, 0) for k, v in dsk.dependents.items() if isinstance(v, set) and len(v) == 0
-    ]
-
```

Comment:
If we drop this line we lose our only-optimize-on-the-first-partition strategy which makes sure the typetracer-only graph only executes on the minimal necessary graph.

Reply:
Ah, yes! I was working on this quite late, and this seems obvious in the light of day. I wonder whether we can determine which of the passed-in keys are collections that we can slice like this?